data/topic-filter.go

```diff
@@ -0,0 +1,213 @@
+		HealthDisabilityAndUnpaidCare.Group:          HealthDisabilityAndUnpaidCare,
+		Housing.Group:                                Housing,
+		LabourMarketAndTravelToWork.Group:            LabourMarketAndTravelToWork,
+		SexualOrientationAndGenderIdentity.Group:     SexualOrientationAndGenderIdentity,
+		Veterans.Group:                               Veterans,
+	}
+)
+
+// reviewFilter retrieves filters from query, checks if they are one of the filter options, and updates validatedQueryParams
+func reviewTopicFilters(ctx context.Context, urlQuery url.Values, validatedQueryParams *SearchURLParams) error {
+	topicFilters := urlQuery["topics"]
+
+	if topicFilters == nil {
+		return nil
+	}
+
+	topics := strings.Split(topicFilters[0], ",")
+
+	for _, topicFilterQuery := range topics {
+
+		topicFilterQuery = strings.ToLower(topicFilterQuery)
+
+		if topicFilterQuery == "" {
+			continue
+		}
+
+		topicFilter, found := topicFilterOptions[topicFilterQuery]
+
+		if !found {
+			err := errs.ErrFilterNotFound
+			logData := log.Data{"topic filter not found": topicFilter}
+			log.Error(ctx, "failed to find topic filter", err, logData)
+
+			return err
+		}
+
+		validatedQueryParams.TopicFilter.Query = append(validatedQueryParams.TopicFilter.Query, topicFilter.Group)
+		validatedQueryParams.TopicFilter.LocaliseKeyName = append(validatedQueryParams.TopicFilter.LocaliseKeyName, topicFilter.LocaliseKeyName)
+	}
+
+	return nil
+}
+
+// GetTopicCategories returns all the categories and its content types where all the count is set to zero
+func GetTopicCategories() []TopicCategory {
+	var topicCategories []TopicCategory
+	topicCategories = append(topicCategories, TopicCategories...)
+
+	// To get a different reference of Topic - deep copy
+	for i, topicCategory := range topicCategories {
+		topicCategories[i].Topics = []Topic{}
+		topicCategories[i].Topics = append(topicCategories[i].Topics, TopicCategories[i].Topics...)
+
+		// To get a different reference of SubTypes - deep copy
+		for j := range topicCategory.Topics {
+			topicCategories[i].Topics[j].SubTopics = []string{}
+			topicCategories[i].Topics[j].SubTopics = append(topicCategories[i].Topics[j].SubTopics, TopicCategories[i].Topics[j].SubTopics...)
+		}
+	}
+
+	return topicCategories
+}
+
+// updateQueryWithAPITopics retrieves and adds all available subtopics which is related to the search filter given by the user
+func updateQueryWithAPITopics(apiQuery url.Values) {
+	topics := apiQuery["topics"]
+
+	if len(topics) > 0 {
+		subTopics := getListOfSubTopics(topics)
+
+		apiQuery.Set("topics", strings.Join(subTopics, ","))
+	} else {
+		apiQuery.Set("topics", defaultTopics)
+	}
+}
+
+// getListOfSubTopics gets all available subtopics for topics which have been set to filter results by the user
+func getListOfSubTopics(topics []string) []string {
+	var subTopics []string
+
+	for _, topic := range topics {
+		subFilter := topicFilterOptions[topic]
+		subTopics = append(subTopics, subFilter.SubTopics...)
```

Comment:
As mentioned above, we will want to change the `subFilter` variable name to `topicFilterOption`:

```suggestion
		subTopics = append(subTopics, topicFilterOption.SubTopics...)
```